### PR TITLE
Add request domain support

### LINKED ISSUE #35: Add request site support
as explained in https://github.com/SexualHealthInnovations/django-wizard-builder/pull/30#issuecomment-293377966, that PR was changed to add only basic SITE_ID support. wizard builder still needs more advanced site support, enabled via passing a request / site from the view down to the site aware queries

### PR DESCRIPTION
**Pull request by lynnco**
closes #35

### CODE REVIEW

**Comment by lynnco** [comment]
my fancy new way to test request domain support is to remove the SITE_ID, and get all the tests to pass that way

**Comment by lynnco** [comment]
I may want to try this with https://github.com/SexualHealthInnovations/callisto-core ? I dunno

File: tests/test_app/views.py [hunk 1]
```diff
@@ -25,12 +26,28 @@ def done(self, form_list, **kwargs):
...
+    return TestWizard.wizard_factory(
+        site_id=site.id,
+        object_to_edit=report,
+    ).as_view(
+        url_name="test_edit_wizard",
+        template_name='wizard_form.html',
+    )(
+        request,
+        step=step,
+    )
```
Comment (lynnco) [inline tests/test_app/views.py]: actual diff:

```diff
+ site = get_current_site(request)
+ site_id=site.id,
```

**Comment by lynnco** [comment]
If `wizard_factory` is the main entry point for interacting with wizard builder, this PR might be done???

(pending me combing edge case determining any needed tests)

File: tests/test_models.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def test_questions_get_added_to_end_by_default(self):
         for i in range(pages):
             QuestionPage.objects.create()
         question = SingleLineText.objects.create(text="This is a question with no page")
-        self.assertEqual(question.page.position, pages+1)
+        self.assertEqual(question.page.position, pages + 1)
```
Comment (lynnco) [inline tests/test_models.py]: not sure why a previous PR didnt catch this *shrug*, but I run

```bash
#!/usr/bin/env bash

isort -rc tests/
isort -rc wizard_builder/
autopep8 --in-place --recursive --aggressive --aggressive tests/ --max-line-length 119 --exclude="*/migrations/*"
autopep8 --in-place --recursive --aggressive --aggressive wizard_builder/ --max-line-length 119 --exclude="*/migrations/*"
flake8 tests/
flake8 wizard_builder/
```

locally

**Comment by lynnco** [comment]
Actually, I'm going to need to test callisto-core (and private callisto) without SITE_ID set, that's the only way to test request support

**Comment by lynnco** [comment]
^ but that doesn't effect this PR, so this PR is up for review! @kelseyq

File: tests/test_sites.py [hunk 3]
```diff
@@ -22,16 +22,18 @@ def __exit__(self, *args):
...
+        with TempSiteID(1):
+            page = QuestionPage.objects.create()
         self.assertEqual(page.site.domain, 'example.com')
 
     def test_on_site_respects_SITE_ID_setting(self):
         site_1_pages = 3
         site_2_pages = site_1_pages + 1
-        site_2 = Site.objects.create()
+        with TempSiteID(1):
+            site_2 = Site.objects.create()
```
Comment (kelseyq) [inline tests/test_sites.py]: what difference does making the Site object with a specific SITE_ID setting make?

Comment (lynnco) [inline tests/test_sites.py]: oh oops, it doesn't make a difference

**Comment by kelseyq** [comment]
is there already an end-to-end test that shows that a request to wizard_factory with a given site doesn't show pages from another site?

File: tests/test_sites.py (hunk 3 above)
Comment (lynnco) [inline tests/test_sites.py]: I found another issue in this PR anyways (from working on this https://github.com/SexualHealthInnovations/callisto-core/commit/beec2e70a36c8fdfe924b590d68b95c709e50caa)

**Comment by lynnco** [comment]
no, the SITE_ID test you were just commenting on does that though

I can add the request domain test that callisto-core has

**Comment by kelseyq** [comment]
yeah if it's straightforward let's do that!

**Comment by lynnco** [comment]
I tested this with https://github.com/SexualHealthInnovations/callisto-core/pull/163 locally, and it looks good to go. So I'm going to make the PR version bump to 0.0.9

**Comment by lynnco** [comment]
up for review!
